model-api-gen-gradle/src/main/kotlin/org/modelix/metamodel/gradle/GenerateAntScriptForMpsMetaModelExport.kt

```diff
@@ -1,31 +1,28 @@
 package org.modelix.metamodel.gradle
 
 import org.gradle.api.DefaultTask
-import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.model.ObjectFactory
 import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.*
 import org.gradle.api.tasks.Optional
+import java.io.File
 import java.util.*
 import javax.inject.Inject
 
 @CacheableTask
 abstract class GenerateAntScriptForMpsMetaModelExport @Inject constructor(of: ObjectFactory) : DefaultTask() {
-    @get:InputDirectory
-    @PathSensitive(PathSensitivity.RELATIVE)
-    val mpsHome: DirectoryProperty = of.directoryProperty()
+    @Input
+    val mpsHome: Property<String> = of.property(String::class.java)
 
     @get:OutputFile
     val antScriptFile: RegularFileProperty = of.fileProperty()
 
-    @InputDirectory
-    @PathSensitive(PathSensitivity.RELATIVE)
+    @Input
     val exporterDir: Property<String> = of.property(String::class.java)
 
-    @InputFiles
-    @PathSensitive(PathSensitivity.RELATIVE)
+    @Input
```

Comment:
What's the reason behind these changes?

Reply:
The task `GenerateAntScriptForMpsMetaModelExport` had directories and the containing files as inputs.

But generating the Ant script only needed the directory names as inputs. Therefore, it was executed more often the necessary.

Reply:
`@InputFiles`/`@ InputDirectory` marks the file/directory name and the content of the file/directory as input . 
But the task does not use the content of the directories and files.

> Note: To make the task dependent on the directory's location but not its contents, expose the path of the directory as an [Input](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Input.html) property instead.

see. https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/InputDirectory.html

> This will cause the task to be considered out-of-date when the property has changed. This annotation cannot be used on a File object. If you want to refer to the file path, independently of its contents, r**eturn a String instead which returns the absolute path**. If, instead, you want to refer to the contents and path of a file or a directory, use [InputFile](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/InputFile.html) or [InputDirectory](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/InputDirectory.html) respectively.

see also. https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Input.html